src/stk500v2.c

```diff
@@ -1306,7 +1314,7 @@ static int stk500v2_initialize(const PROGRAMMER *pgm, const AVRPART *p) {
       stk500v2_getparm(pgm, PARAM_OSC_CMATCH, &osc_cmatch);
       if(osc_pscale) {
         int prescale = 1;
-        f_get = STK500V2_XTAL / 2;
+        f_get = PDATA(pgm)->xtal / 2;
```

Comment:
Not a fault of this PR, but `osc_pscale` and `osc_cmatch` ought to be set to 0 above b/c `stk500v2_getparm()` might not set them! Ideally, *also* the return value of `stk500v2_getparm()` calls ought to be checked to guard against not setting the `osc_...` variables.

Reply:
This will be a bigger separate commit, because `stk500v2_getparm()` (as well as `stk500_getparm()` in `stk500.c`) is used quite often w/o checking the return value. Nevertheless I can *adopt* this issue and fix it in the course of my stk500* overhaul.

Reply:
Thanks for cleaning this in your overhaul of stk500* - much appreciated.